eng/templates/job/build-from-source.yml

```diff
@@ -0,0 +1,96 @@
+parameters:
+  # Specifies the prefix for source-build jobs added to pipeline. Use this if disambiguation is necessary.
+  jobNamePrefix: 'sourcebuild'
+
+  # Specifies that the repo only builds managed components. This indicates that output is not RID-specific 
+  # and only one build job is required. A linux x64 machine will be used, but the build must continue to 
+  # work on all supported platforms equally.
+  managedOnly: false
+  
+  # Specifies that the repo has native components that can link to platform-provided libraries.
+  #   - The job passes /p:ArcadeBuildNonPortable=true to the build to enable non-portable behavior.
+  #   - The job passes a platform-specific TargetRid MSBuild property.
+  #   - Setting this parameter adds additional platform-specific "non-portable" build jobs.
+  includeNonPortable: false
```

Comment:
When I went through the repos, they were all `managedOnly:true` or `managedOnly:false includeNonPortable:true`--I think we can remove `includeNonPortable`. I'm not actually sure when it would ever be used--if someone's building native code, distros prefer to build it non-portable.

Reply:
I refactored, taking this in mind.